lib/models/entitiesStore.js

```diff
@@ -28,15 +28,89 @@ var async = require('async'),
     entitiesCollectionName = require('../../config').orionDb.collection,
     myutils = require('../myutils'),
     constants = require('../constants'),
-    logger = require('logops');
+    logger = require('logops'),
+    NGSI = require('ngsijs');
 
 function orionServiceDb(service) {
     return appContext.OrionDb(config.orionDb.prefix + '-' + service);
 }
 
-function findSilentEntities(service, subservice, ruleData, func, callback) {
+function findSilentEntitiesByAPI(service, subservice, ruleData, func, callback) {
+    var context = { op: 'checkNoSignal.findSilentEntitiesByAPI', comp: constants.COMPONENT_NAME };
+
+    var options = {
+        service: service,
+        servicepath: subservice
+    };
+    options.headers = {};
+    // if (token !== null) {
+    //     options.headers[constants.AUTH_HEADER] = token;
+    // }
+    // Add correlator
+    var domain = process.domain;
+    if (domain && domain.context) {
+        options.headers[constants.CORRELATOR_HEADER] = domain.context.corr;
+        // Add other headers
+        if (domain.context.srv && options.headers[constants.SERVICE_HEADER] === undefined) {
```

Comment:
What do you think about making it more explicit (easy to understand):
(domain.context.srv && (options.headers[constants.SERVICE_HEADER] === undefined))

It took me a while.